libp2p/services/autorelayservice.nim

```diff
@@ -41,7 +41,7 @@ proc addressMapper(
 
 proc reserveAndUpdate(
     self: AutoRelayService, relayPid: PeerId, switch: Switch
-) {.async.} =
+) {.async: (raises: [CatchableError]).} =
```

Comment:
`CatchableError` used to simplify raised errors here, as there could be many different errors raised but caller don't really care what is cause of error.